src/mds/Server.cc

```diff
@@ -2256,6 +2260,10 @@ void Server::reply_client_request(MDRequestRef& mdr, const ref_t<MClientReply> &
     mds->send_message_client(reply, session);
   }
 
+  if (client_inst.name.is_mds() && reply->get_op() == CEPH_MDS_OP_RENAME) {
+    mds->send_message(reply, mdr->client_request->get_connection());
```

Comment:
Hmmm.. the reintegration (rename) operation was never replied back to the peer (request initiator) mds.